src/main/java/rosenfeld/boggle/BoggleSolver.java

```diff
@@ -52,6 +52,7 @@ private void boardSearch (List<Location> neighbors, String currPrefix, boolean [
                 List<Location> newNeighbors = getNeighbors(neighbor.getRow(), neighbor.getCol(), visited);
                 boardSearch(newNeighbors, currWord, visited, boggleMatrix);
             }
+            visited[neighbor.getRow()][neighbor.getCol()] = false;
```

Comment:
Here was the bug that caused the solver to find very few words - I never reset a neighbor's "visited" to false after finding no more possibilities, so there were increasingly fewer possibilities of choices on the board.